### PR TITLE
Feature/add tls

### PR DESCRIPTION
**Pull request by paulomach**
# Issue

TLS must be on the bundle default deployment.

# Solution

Add TLS to bundle.

# To fix
- add/enable tests

### CODE REVIEW

File: tests/integration/test_bundle.py [hunk 1]
```diff
@@ -8,9 +8,14 @@
...
 
-async def test_placeholder(ops_test: OpsTest) -> None:
-    """A placeholder for an integration test to be implemented."""
-    assert True
+async def test_deploy_bundle(ops_test: OpsTest) -> None:
+    """Deploy bundle."""
+    async with ops_test.fast_forward():
+        await ops_test.model.deploy("./releases/latest/mysql-k8s-bundle.yaml", trust=True)
+
+        await ops_test.model.wait_for_idle(apps=[MYSQL_APP, ROUTER_APP, TLS_APP], timeout=5 * 60)
```
Comment (shayancanonical) [inline tests/integration/test_bundle.py:21]: should we assert that the apps reach a certain status? (even if it is `waiting`)

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -17,21 +17,3 @@ jobs:
         run: python3 -m pip install tox
       - name: Run linters
         run: tox -e lint
-
```
Comment (MiaAltieri) [inline .github/workflows/ci.yaml:20]: ugh I am a noob, why is this removed?

File: releases/latest/mysql-k8s-bundle.yaml [hunk 3]
```diff
@@ -1,20 +1,34 @@
...
+source: https://github.com/canonical/mysql-k8s-bundle
+issues: https://github.com/canonical/mysql-k8s-bundle/issues
 applications:
   mysql-k8s:
     channel: latest/edge
     charm: mysql-k8s
     constraints: arch=amd64
-    revision: 17
-    scale: 1
+    revision: 19
```
Comment (MiaAltieri) [inline releases/latest/mysql-k8s-bundle.yaml:11]: ill trust you that you're bumping the revision by 2 here 😉

File: releases/latest/mysql-k8s-bundle.yaml [hunk 4]
```diff
@@ -1,20 +1,34 @@
...
-name: mysql-k8s-bundle
+  tls-certificates-operator:
+    channel: latest/edge
+    charm: tls-certificates-operator
+    constraints: arch=amd64
+    revision: 11
+    scale: 1
+    options:
+      ca-common-name: canonical
+      generate-self-signed-certificates: true
```
Comment (MiaAltieri) [inline releases/latest/mysql-k8s-bundle.yaml:28]: might be worth mentioning somewhere that in the bundle you default to self signed certs when using TLS, (see suggestion in README)

File: README.md [hunk 5]
```diff
@@ -21,9 +22,10 @@ Finally add a juju model with `juju add-model <model-name>` and deploy the bundl
 ## Bundle Components
 - [mysql-k8s](https://charmhub.io/mysql-k8s): A k8s charm to deploy MySQL with Group Replication.
 - [mysql-router-k8s](https://charmhub.io/mysql-router-k8s) - a k8s charm to deploy MySQL Router.
+- [tls-certificates-operator](https://charmhub.io/tls-certificates-operator) - TLS operator.
```
Comment (MiaAltieri) [inline README.md:25]:
```suggestion
- [tls-certificates-operator](https://charmhub.io/tls-certificates-operator) - TLS operator. Note: The TLS settings in bundles use self-signed-certificates which are not recommended for production clusters, the tls-certificates-operator charm offers a variety of configurations, read more on the TLS charm [here](https://charmhub.io/tls-certificates-operator).
```

File: tests/integration/test_bundle.py [hunk 6]
```diff
@@ -8,9 +8,18 @@
...
-    assert True
+async def test_deploy_bundle(ops_test: OpsTest) -> None:
+    """Deploy bundle."""
+    async with ops_test.fast_forward():
+        await ops_test.model.deploy("./releases/latest/mysql-k8s-bundle.yaml", trust=True)
+
+        await ops_test.model.wait_for_idle(
+            apps=[MYSQL_APP, TLS_APP], status="active", timeout=5 * 60
+        )
+
```
Comment (MiaAltieri) [inline tests/integration/test_bundle.py:24]:
```suggestion
# we except the router to be in waiting status because <reason>
```

**Review by MiaAltieri — COMMENTED** [review]
Looks awesome! Minor comments only

**Review by marceloneppel — APPROVED** [review]
LGTM!

**Review by WRFitch — APPROVED** [review]
LGTM once Mia's comments are resolved. Specifically, it's worth documenting our certs are self-signed.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (paulomach) [reply .github/workflows/ci.yaml:20]: as for vm, there's no tests in place - this would just fail CI now

File: README.md (hunk 5 above)
Comment (paulomach) [reply README.md:25]: lovely!

File: releases/latest/mysql-k8s-bundle.yaml (hunk 3 above)
Comment (paulomach) [reply releases/latest/mysql-k8s-bundle.yaml:11]: since the PR is up, the charm is being updated, so it may even be higher

File: releases/latest/mysql-k8s-bundle.yaml (hunk 4 above)
Comment (paulomach) [reply releases/latest/mysql-k8s-bundle.yaml:28]: suggestion commited

**Review by shayancanonical — APPROVED** [review]
